src/porepy/fracs/simplex.py

```diff
@@ -82,20 +84,21 @@ def tetrahedral_grid(fracs=None, box=None, network=None, **kwargs):
             # isinstance-statements, but that became less than elegant. To
             # revisit.
             if hasattr(f, 'p') and isinstance(f.p, np.ndarray):
-                # Convert the fractures from numpy representation to our 3D
-                # fracture data structure.
                 frac_list.append(f)
             else:
+                # Convert the fractures from numpy representation to our 3D
+                # fracture data structure. 
                 frac_list.append(fractures.Fracture(f))
 
         # Combine the fractures into a network
         network = fractures.FractureNetwork(frac_list, verbose=verbose,
                                             tol=kwargs.get('tol', 1e-4))
-
+    # Add any subdomain boundaries:
+    network.add_subdomain_boundaries(subdomains)
```

Comment:
I suppose it is critical that subdomains are added before boundaries? If so, this should be documented, but probably in FractureNetwork

In the longer term, we should aim for a Fracture network implementation that allows us to add and remove fractures, and generally play around, without everything having to be recomputed. But that is for another day.